### PR TITLE
Update submitter design doc to match impl

### PR DESCRIPTION
**Pull request by CLEckhardt**
Adds implmentation details and diagrams to the batch submitter design
doc.

Signed-off-by: Chris Eckhardt <eckhardt@bitwise.io>

### CODE REVIEW

File: community/planning/batch_submitter.md [hunk 1]
```diff
@@ -8,470 +11,354 @@
...
+        task3
+        ...
+    end
+    B -->|submission| task1 & task2 & task3 & ...
+    task1 & task2 & task3 & ... -->|response| C[response collector]
+    C -->|notifications| D[observer]
+    B -->|notifications of submission starts| D
+</div>
+
+It the above diagram, the poller continuosly polls the queue for a new
```
Comment (ameliabradley) [inline community/planning/batch_submitter.md:60]: continuosly -> continuously

File: community/planning/batch_submitter.md [hunk 2]
```diff
@@ -8,470 +11,354 @@
...
+submission and sends the submission to the async runtime. If the queue does not
+have a new submission ready, the poller will pause for a predetermined interval
+before polling again.
+
+The async runtime drives the execution of the async POST requests to the DLT.
+Each submission is spawned in its own task handler, which controlls retry
```
Comment (ameliabradley) [inline community/planning/batch_submitter.md:66]: controlls -> controls

File: community/planning/batch_submitter.md [hunk 3]
```diff
@@ -8,470 +11,354 @@
...
-interpretation. The mechanism could be a method on a store, a call to a REST
-API, method calls on multiple other observers, etc. The implementation of the
-observer also determines how the observer interprets and acts on the
-notification.
+A new submission task is spawned for every batch submission. These run in the
+async runtime and complete concurrently.
 
-Here is the observer implementation in Grid:
+The submission task has three nested subcomponents, which are organized by the
+actions the task needs to complete. They are the task handler (labled "Async
```
Comment (ameliabradley) [inline community/planning/batch_submitter.md:216]: labled -> labeled

File: community/planning/batch_submitter.md [hunk 4]
```diff
@@ -8,470 +11,354 @@
...
+
+When `stop()` is called, the submitter uses messages to wind down its threads.
+It also uses a mutex-guarded `Collector` struct within the `RunningSubmitter`
+to collect the queue, submitter observer, and configured submission command
+factory, rebuilding and returning a `RunnableSubmitter` identical to the one
+that spawned it. Calling `run()` on this `RunnableSubmitter` restarts the
+submission service.
+
+__Note__: Stopping the submitter does not stop the submissions in progress -
+these will continue to execute to completion before the subsission service
```
Comment (ameliabradley) [inline community/planning/batch_submitter.md:302]: subsission -> submission

**Review by ameliabradley — CHANGES_REQUESTED** [review]

File: community/planning/batch_submitter.md (hunk 1 above)
Comment (CLEckhardt) [reply community/planning/batch_submitter.md]: fixed, thanks!

File: community/planning/batch_submitter.md (hunk 2 above)
Comment (CLEckhardt) [reply community/planning/batch_submitter.md]: fixed, thanks!

File: community/planning/batch_submitter.md [hunk 5]
```diff
@@ -8,470 +11,354 @@
...
+</div>
+
+It the above diagram, the poller continuosly polls the queue for a new
+submission and sends the submission to the async runtime. If the queue does not
+have a new submission ready, the poller will pause for a predetermined interval
+before polling again.
+
+The async runtime drives the execution of the async POST requests to the DLT.
+Each submission is spawned in its own task handler, which controlls retry
+behavior and collects relavent information on the submission process through
```
[the remark this answers is not in the document]
Comment (CLEckhardt) [reply community/planning/batch_submitter.md]: fixed, thanks!

File: community/planning/batch_submitter.md [hunk 6]
```diff
@@ -8,470 +11,354 @@
...
 
-impl BatchAddresser {
-    fn new(base_url: &'static str, parameter: Option<&'static str>) -> Self {
-        Self {
-            base_url,
-            parameter,
-        }
-    }
-}
+This is a generic component that recieves notifications from the submitter. Its
```
[the remark this answers is not in the document]
Comment (CLEckhardt) [reply community/planning/batch_submitter.md]: fixed, thanks!

File: community/planning/batch_submitter.md (hunk 3 above)
Comment (CLEckhardt) [reply community/planning/batch_submitter.md]: fixed, thanks!

**Review by ameliabradley — CHANGES_REQUESTED** [review]

File: community/planning/batch_submitter.md [hunk 7]
```diff
@@ -8,470 +11,354 @@
...
+    Async Task ->>- Listener/Observer: submission response via `notify()`
+</div>
+
+There are a number of high-level parts that comprise the submitter
+implementation:
+
+#### Submission
+
+This struct represents a batch to be submitted. It is how the batch payload and
+relevent data are passed through the submitter and is what the queue delivers
```
[the remark this answers is not in the document]
Comment (CLEckhardt) [reply community/planning/batch_submitter.md]: fixed, thanks!

File: community/planning/batch_submitter.md [hunk 8]
```diff
@@ -8,470 +11,354 @@
...
+thread that notifies the observer.
 
-pub struct BatchTrackingObserver {
-    store: Box<dyn BatchTrackingStore>,
-}
+The submission controller controls the retry and response behavior of the
+submission command. For example, if the DLT is busy, the controller will
+continue to execute the submission command until the submission is successful
+or a predetermined number of submission attempts have failed. The controller
+propogates any errors up to the task handler, which communicates them to the
```
[the remark this answers is not in the document]
Comment (CLEckhardt) [reply community/planning/batch_submitter.md]: fixed, thanks!

**Review by ameliabradley — DISMISSED** [review]

**Review by ameliabradley — APPROVED** [review]

**Pull request merged**